QuickFIXn/Fields/Converters/DateTimeConverter.cs

```diff
@@ -9,197 +11,620 @@ namespace QuickFix.Fields.Converters;
     }
 
+    /// <summary>
+    /// Converts the specified <see cref="DateTime"/> to a <see cref="string"/>.
+    /// </summary>
+    /// <param name="dt">The value to convert.</param>
+    /// <param name="precision">The level of precision with which to format the fractional seconds component of <paramref name="dt"/>.</param>
+    /// <returns>
+    /// A value representing <paramref name="dt"/>. The value will begin in the format "yyyyMMdd-HH:mm:ss"
+    /// and end in fractional seconds whose precision is determined by <paramref name="precision"/>.
+    /// </returns>
+    /// <exception cref="ArgumentOutOfRangeException"><paramref name="precision"/> is an invalid value.</exception>
     public static string ToFIX(DateTime dt, TimeStampPrecision precision)
     {
-        return (precision == TimeStampPrecision.Nanosecond)
-            ? string.Format(DATE_TIME_FORMAT_WITH_NANOSECONDS, dt, dt.SubsecondAsNanoseconds())
-            : string.Format(DATE_TIME_PRECISION_TO_FORMAT[precision], dt);
+        return precision switch
+        {
+            TimeStampPrecision.Second => dt.ToString("yyyyMMdd-HH:mm:ss"),
+            TimeStampPrecision.Millisecond => dt.ToString("yyyyMMdd-HH:mm:ss.fff"),
+            TimeStampPrecision.Microsecond => dt.ToString("yyyyMMdd-HH:mm:ss.ffffff"),
+            TimeStampPrecision.Nanosecond => $"{dt:yyyyMMdd-HH:mm:ss}.{SubsecondAsNanoseconds(dt):000000000}",
+            _ => throw new ArgumentOutOfRangeException(nameof(precision)),
+        };
     }
 
-    public static string ToFIXDateOnly(DateTime dt)
+    /// <summary>
+    /// Converts the specified <see cref="DateOnly"/> to a <see cref="string"/> in the format "yyyyMMdd".
+    /// </summary>
+    /// <param name="date">The value to convert.</param>
+    /// <returns>A value representing <paramref name="date"/> in the format "yyyyMMdd".</returns>
+    public static string ToFIXDateOnly(DateOnly date)
     {
-        return string.Format(DATE_ONLY_FORMAT, dt);
+        return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
     }
 
-    public static string ToFIXTimeOnly(DateTime dt, TimeStampPrecision precision)
+    /// <summary>
+    /// Converts the date component of the specified <see cref="DateTime"/> to a <see cref="string"/> in the format "yyyyMMdd".
+    /// </summary>
+    /// <param name="dt">The value to convert.</param>
+    /// <returns>A value representing the date component of <paramref name="dt"/> in the format "yyyyMMdd".</returns>
+    public static string ToFIXDateOnly(DateTime dt) => ToFIXDateOnly(DateOnly.FromDateTime(dt));
+
+    /// <summary>
+    /// Converts the specified <see cref="TimeOnly"/> to a <see cref="string"/> in the format "HH:mm:ss.fff".
+    /// </summary>
+    /// <param name="time">The value to convert.</param>
+    /// <returns>A value representing <paramref name="time"/> in the format "HH:mm:ss.fff".</returns>
+    public static string ToFIXTimeOnly(TimeOnly time)
```

Comment:
holy baby jesus, we have 6 overloads of ToFIXTimeOnly now...

Reply:
The engine uses only `ToFIXTimeOnly(DateTime, TimeStampPrecision)`.

This one then calls `ToFIXTimeOnly(TimeOnly time, TimeStampPrecision precision)`.

So now I'm thinking the rest of them are unnecessary.